src/com/sap/piper/ConfigurationHelper.groovy

```diff
@@ -51,7 +49,7 @@ class ConfigurationHelper implements Serializable {
         return mixin(stepConfiguration, filter, compatibleParameters)
     }
 
-    final ConfigurationHelper mixin(Map parameters, Set filter = null, Map compatibleParameters = [:]){
```

Comment:
How does the modifier `final` relate to `@NonCPS`?

Reply:
I cannot really explain what happens here yet. I've seen those log lines

```
[2019-08-12T11:10:53.741Z] --- Begin library step of: artifactSetVersion ---
[Pipeline] sh
[2019-08-12T11:10:54.043Z] + git rev-parse --is-inside-work-tree
[Pipeline] sh
[2019-08-12T11:10:54.330Z] + git diff --quiet HEAD
[Pipeline] echo
[2019-08-12T11:10:54.365Z] 
[Pipeline] sh
[2019-08-12T11:10:54.678Z] + git rev-parse --is-inside-work-tree
[Pipeline] sh
[2019-08-12T11:10:54.967Z] + git rev-parse HEAD
expected to call java.util.LinkedHashMap.mixin but wound up catching org.jenkinsci.plugins.workflow.cps.CpsClosure2.call; see: https://jenkins.io/redirect/pipeline-cps-method-mismatches/
expected to call java.util.LinkedHashMap.mixin but wound up catching org.jenkinsci.plugins.workflow.cps.CpsClosure2.call; see: https://jenkins.io/redirect/pipeline-cps-method-mismatches/
[Pipeline] sh
```

I traced back what code is behind, and came here. Wondered about the `final`, removed it, warnings disappeared. What does the `final` do anyway? Looks like it got here by mistake, or does it have a purpose?

Reply:
I can't explain the purpose of the final here. In case this is the reason for `@NonCPS` issues we can remove it. Was just curiosity. Also can't explain why this triggers on `@NonCPS` issue ...

Reply:
I'll try to reproduce it in a minimal example later to better understand what went wrong, as soon as I have some time to spare.